Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -108,6 +117,34 @@ var JSTimersExecution = {
     }
   },
 
+  callIdleCallbacks: function(frameTime) {
+    // Arbitrary threshold: 10ms remaining in frame.
+    if (17 - (performanceNow() - frameTime) < 10) {
```

Comment:
Taking time at the beginning of the frame (by overrunning into the next frame) seems pretty dangerous to me. I mean, generally trying to schedule at the granularity of half-a-frame without hooks into the actual scheduler seems dangerous to me, but *assuming* you're going to have idle time is even more so :P